src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +567,170 @@ subroutine DustEmission (bounds, &
-            ! purpose: adjust threshold friction velocity to acct for moisture and
-            !          roughness. The ratio tmp1 / sqrt(forc_rho) comes from
-            !          subr. wnd_frc_thr_slt_get which computes dry threshold
-            !          friction velocity for saltation
+         else
+            wnd_frc_slt = fv(p)                     ! The value here is not important since once lnd_frc_mbl(p) <= 0.0_r8 there will be no emission.
+            frc_thr_rghn_fct(p) = 0.0_r8            ! save and output hybrid drag partition factor
+         end if
+
+         !########## end of drag partition effect #######################################################
+
+         ! save soil friction velocity and roughness effect before the if-statement
+         wnd_frc_soil(p) = wnd_frc_slt  ! save soil friction velocity for CLM output, which has drag partition and Owen effect
+         ! save land mobile fraction
+         lnd_frc_mble(p) = lnd_frc_mbl(p)  ! save land mobile fraction first, before the if-statement
+         ! only perform the following calculations if lnd_frc_mbl is non-zero
 
-            wnd_frc_thr_slt = tmp1 / sqrt(forc_rho(c)) * frc_thr_wet_fct * frc_thr_rgh_fct
+         if (lnd_frc_mbl(p) > 0.0_r8) then  ! if bare land fraction is larger than 0 then calculate the dust emission equation
 
             ! reset these variables which will be updated in the following if-block
 
-            wnd_frc_slt = fv(p)
             flx_mss_hrz_slt_ttl = 0.0_r8
             flx_mss_vrt_dst_ttl(p) = 0.0_r8
 
             ! the following line comes from subr. dst_mbl
             ! purpose: threshold saltation wind speed
 
-            wnd_rfr_thr_slt = u10(p) * wnd_frc_thr_slt / fv(p)
+            wnd_rfr_thr_slt = u10(p) * wnd_frc_thr_slt / fv(p)     ! keep and use if we want the default Z03 scheme
+
+            ! the following comes from subr. flx_mss_hrz_slt_ttl_Whi79_get
+            ! purpose: compute vertically integrated streamwise mass flux of particles
+
+            if (wnd_frc_slt > wnd_frc_thr_slt_it) then! if using Leung's scheme, use impact threshold for dust emission equation
+
+               !################### for Leung et al. (2022) ################################################
+               !################ uncomment the below block if want to use Leung's scheme ###################
 
-            ! the following if-block comes from subr. wnd_frc_slt_get 
-            ! purpose: compute the saltating friction velocity
-            ! theory: saltation roughens the boundary layer, AKA "Owen's effect"
+               flx_mss_vrt_dst_ttl(p) = dst_emiss_coeff(p) * mss_frc_cly_vld(c) * forc_rho(c) * ((wnd_frc_slt**2.0_r8 - wnd_frc_thr_slt_it**2.0_r8) / wnd_frc_thr_slt_std) * (wnd_frc_slt / wnd_frc_thr_slt_it)**frag_expt  ! Leung et al. (2022) uses Kok et al. (2014) dust emission euqation for emission flux
 
-            if (u10(p) >= wnd_rfr_thr_slt) then
```

Comment:
Erik: Just want to note here that Owen's effect should be kept in Zender's scheme, and not in Leung's scheme. So please save this for Zender. Thanks

Reply:
OK, I'll remove it here since we are preserving Zender's scheme.